File: backend/Models/Booking.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models;

public class Booking
{
    public int BookingId { get; set; }
    public required bool IsActive { get; set; }
    public DateTime BookingDate { get; set; }
    public DateTime EndDate { get; set; }

    [ForeignKey("UserId")]
    public string UserId { get; set; } = string.Empty;
    public ApplicationUser? User { get; set; }

    [ForeignKey("ResourceId")]
    public Resource? Resource { get; set; }
    public int ResourceId { get; set; }
}

File: frontend/src/types/booking.ts
export interface Booking {
  bookingId: number;
  bookingDate: string;
  endDate: string;
  isActive: boolean;
  timeslot: "FM" | "EM";

  user: {
    id: string;
    email: string;
    firstName?: string;
    lastName?: string;
    role?: string;
  };

  userId?: string;

  resource: {
    resourceId: number;
    name: string;
    isBooked: boolean;
    resourceType?: {
      resourceTypeId: number;
      name: string;
    };
  };
}

export interface BookingDTO {
  resourceId: number;
  bookingDate: string;
  timeslot: "FM" | "EM";
}